src/main/java/nl/praegus/fitnesse/responders/symbolicLink/SymbolicLinkResponder.java

```diff
@@ -0,0 +1,124 @@
+        // Get Symlinks of the wiki page
+        WikiPageProperty symLinksProperty = pageData.getProperties().getProperty(SymbolicPage.PROPERTY_NAME);
+
+        // Loop through the names
+        if (symLinksProperty != null) {
+            for (String name : symLinksProperty.keySet()) {
+                symbolicLinkArray.put(getObject(name, wikiPage, symLinksProperty));
+            }
+        }
+
+        // Get Symlinks from the children
+        if (getPageChildren(wikiPage).size() > 0) {
+            for (WikiPage p : getPageChildren(wikiPage)) {
+                getSymlinkHelper(p, symbolicLinkArray);
+            }
+        }
+
+        return symLinksProperty != null ? symbolicLinkArray : new JSONArray();
+    }
+
+    private JSONObject getObject(String name, WikiPage wikiPage, WikiPageProperty symLinksProperty) {
+        JSONObject symboliclinkInformation = new JSONObject();
+        symboliclinkInformation.put("pagePath", wikiPage.getFullPath().toString());
+        symboliclinkInformation.put("linkName", name);
+        symboliclinkInformation.put("linkPath", makePathForSymbolicLink(wikiPage, symLinksProperty.get(name)));
+        symboliclinkInformation.put("backUpLinkPath", symLinksProperty.get(name));
+        return symboliclinkInformation;
+    }
+
+    private String makePathForSymbolicLink(WikiPage page, String linkPath) {
+        WikiPagePath wikiPagePath = PathParser.parse(linkPath);
+
+        if (wikiPagePath != null) {
+            WikiPage parent = wikiPagePath.isRelativePath() ? page.getParent() : page;
+            PageCrawler crawler = parent.getPageCrawler();
+            WikiPage target = crawler.getPage(wikiPagePath);
+            WikiPagePath fullPath;
+            if (target != null) {
+                fullPath = target.getFullPath();
+                fullPath.makeAbsolute();
+            } else {
+                fullPath = new WikiPagePath();
+            }
+            String pathString = fullPath.toString();
+
+            return pathString.length() > 0 && pathString.charAt(0) == '.' ? pathString.substring(1) : pathString;
+        }
+        return null;
+    }
+
+    private String createHTMLPage(FitNesseContext fitNesseContext, Request request, WikiPage wikiPage) {
+        HtmlPage html = fitNesseContext.pageFactory.newPage();
+        html.setNavTemplate("viewNav");
+        html.setMainTemplate("symlinks");
+        html.put("viewLocation", request.getResource());
+        html.put("symlinks", getSymlink(wikiPage));
+        html.setTitle("Symbolic links: " + request.getResource());
+
+        String tags = "";
+        if(pageData != null)  {
+            tags = pageData.getAttribute(SUITES);
+        }
+        html.setPageTitle(new PageTitle("Symbolic links", PathParser.parse(request.getResource()), tags));
+        return html.html(request);
+    }
+
+    private List<WikiPage> getPageChildren(WikiPage parent) {
+        return new ArrayList<>(parent.getChildren());
```

Comment:
parent.getChildren() geeft al een list terug, dus een nieuwe arraylist maken is niet nodig toch? En dan is het een beetje de vraag waarom deze functie bestaat. Je kan tohc ipv deze functie ook gewoon parent.getChildren() aanroepen?

Reply:
parent.getChildren() aangeroepen en de functie weg gehaald